cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -33,55 +29,52 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
-      set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
-    endif()
-    if(PYTHON_INCLUDE_DIRS AND NOT Python_INCLUDE_DIRS)
-      set(Python_INCLUDE_DIRS "${PYTHON_INCLUDE_DIRS}")
+  find_package(Python2 COMPONENTS Interpreter Development NumPy)
+  if(Python2_Development_FOUND)
+    if(NOT Python3_Development_FOUND)
+      # Only Python2 was found, set as main
+      set(PYTHON_EXECUTABLE "${Python2_EXECUTABLE}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_INCLUDE_DIRS "${Python2_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_LIBRARIES "${Python2_LIBRARIES}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_MAJOR "${Python2_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_MINOR "${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_STRING "${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(NUMPY_FOUND ${Python2_NumPy_FOUND} CACHE INTERNAL "" FORCE)
+      set(NUMPY_INCLUDE_DIRS "${Python2_NumPy_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+    else()
+      # Both Python3 and 2 found, set 2 as 'other'
+      set(OTHER_PYTHON_EXECUTABLE "${Python2_EXECUTABLE}" CACHE INTERNAL "" FORCE)
+      set(OTHER_PYTHON_INCLUDE_DIRS "${Python2_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+      set(OTHER_PYTHON_LIBRARIES "${Python2_LIBRARIES}" CACHE INTERNAL "" FORCE)
+      set(OTHER_PYTHON_VERSION_MAJOR "${Python2_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
+      set(OTHER_PYTHON_VERSION_MINOR "${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(OTHER_PYTHON_VERSION_STRING "${Python2_VERSION_MAJOR}_${Python2_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(OTHER_NUMPY_FOUND ${Python2_NumPy_FOUND} CACHE INTERNAL "" FORCE)
+      set(OTHER_NUMPY_INCLUDE_DIRS "${Python2_NumPy_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
     endif()
-    if(PYTHON_LIBRARIES AND NOT Python_LIBRARIES)
-      set(Python_LIBRARIES "${PYTHON_LIBRARIES}")
-    endif()
-    find_package(Python COMPONENTS Interpreter Development REQUIRED)
-    # Search for NumPy, but not required:
-    find_package(Python COMPONENTS NumPy QUIET)
-    # Compat with find_package(PythonInterp), find_package(PythonLibs), find_package(NumPy)
-    set(PYTHON_EXECUTABLE "${Python_EXECUTABLE}" CACHE INTERNAL "" FORCE)
-    set(PYTHON_INCLUDE_DIRS "${Python_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
-    set(PYTHON_LIBRARIES "${Python_LIBRARIES}" CACHE INTERNAL "" FORCE)
-    set(PYTHON_VERSION_STRING "${Python_VERSION}" CACHE INTERNAL "" FORCE)
-    set(PYTHON_VERSION_MAJOR "${Python_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
-    set(PYTHON_VERSION_MINOR "${Python_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
-    set(NUMPY_FOUND ${Python_NumPy_FOUND} CACHE INTERNAL "" FORCE)
-    set(NUMPY_INCLUDE_DIRS "${Python_NumPy_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+  endif()
+
+  if(NOT Python3_Development_FOUND AND NOT Python2_Development_FOUND)
+    message(FATAL_ERROR "No Python 2 or 3 were found")
```

Comment:
Can we add here a message what should be installed (to help a user to find quicker solution)?